### ISSUE TITLE
Support routing with Webix layout

### ISSUE THREAD

**Issue by ghost** [dandv/meteor-webix#3]
Hi Dan,

just wanted to let you know that I currently look for a solution for that. I can contribute the solution to this repository if you like.

What I have done:
- Created a webix-meteor-templates package that wraps a slightly modified version of https://github.com/webix-hub/webix-meteor-example/blob/master/mtest.html#L52-L72 (Blaze template rendering in Webix)
- Created a layout with Webix with a main region that uses a blaze template with `{{> Router}}` for iron:router

**Comment by dandv** [comment]
Hi Jonas,

Thanks for the offer to help; PRs would be appreciated.

As for where exactly to contribute the code, I hope that @mkozhukh will help unify the repositories. Right now we have three repos:
- https://github.com/dandv/meteor-webix - the Meteor wrapper package. Uses [webix](https://github.com/webix-hub/tracker) and [a fork of webix-meteor-data](https://github.com/dandv/webix-meteor-data).
- https://github.com/dandv/webix-meteor-data - fork of the Meteor proxy (until the Webix team accepts [the PRs](https://github.com/webix-hub/webix-meteor-data/pulls)). Used by the package.
- https://github.com/webix-hub/webix-meteor-example with a standalone example that doesn't use the package but provides extra Meteor integration (the Blaze template rendering)

My suggestions:
1. Seems best to have a single repository as the source for the integration, and it seems easiest to [merge the Meteor integration code from webix-meteor-example into webix-meteor-data](https://github.com/webix-hub/webix-meteor-example/issues/6), or even better, directly into this repo, https://github.com/dandv/meteor-webix, because it does packaging according to the Meteor standard. I've already added @mkozhukh as a contributor to this repo, and I'm happy to transfer it into the webix-hub organization. Also, this repo got quite popular (220+ stars, almost 100 watchers - much more popular than Webix itself!) so we should use that to attract more contributors.
2. That will eliminate the need for https://github.com/dandv/meteor-webix to include https://github.com/webix-hub/webix-meteor-data as a submodule, so the latter repository can be deleted/deprecated and instead we can have just _one_ webix-meteor adapter `.js` file, with the proxy (now in webix-meteor-data) and Blaze template rendering (now in webix-meteor-example).
3. This repo already has an example, so we can merge any extra example code from webix-meteor-example into it as well.

@mkozhukh, what do you think?

**Comment by ghost** [comment]
I think the pro version of Webix has more features and the additional components (pivot table etc.). I'm not sure if the additional features are stripped out from the webix-hub/tracker repository.

So I would suggest to have a separate Webix wrapper package and one or more additional Webix Meteor integration packages that depend on the Webix wrapper package.
Users of the pro version could clone this wrapper package as local package and replace the webix files with the pro version.

I've done this structure for the current project I work on already:

![bildschirmfoto 2015-02-19 um 23 52 25](https://cloud.githubusercontent.com/assets/727955/6277932/07764aec-b893-11e4-8398-6df6b4122212.png)

@mkozhukh Is the https://github.com/webix-hub/webix-meteor-example also under MIT license? The license is missing right now.

**Comment by dandv** [comment]
@Sanjo: good point re. the Pro version. This repo is indeed structured to include Webix and the adapter as two submodules, and a Pro user could overwrite `webix_debug.js` in the webix submodule with the Pro version.

If the webix-meteor proxy adapter needs any extra code for the Pro widgets, that code could still go in the same adapter file, since it will be relatively very small by comparison to the size of the library itself (unless the adapter code somehow needs any part of the Pro version to be already instantiated, which I doubt is the case).

In any case, we can at least unify webix-meteor-data and webix-meteor-example, since the latter includes everything in the former, and fixes the example.

**Comment by mkozhukh** [comment]
> Seems best to have a single repository as the source for the integration

I'm agree about merging webix-meteor-data and webix-meteor-example in this repo.

> Is the https://github.com/webix-hub/webix-meteor-example also under MIT license

Yes. All projects except of main Webix repo are under MIT license. I will update readme files to make it clear. 

> If the webix-meteor proxy adapter needs any extra code for the Pro widgets

While PRO version has some extra widgets it uses the same code for data loading as GPL version. There will be no need for PRO-related modifications in data-adapter.

**Comment by mkozhukh** [comment]
About "routing with Webix layout"

As far as I can see there are two possible solution:
- "Meteor first" app. In such app main UI is created through Meteor templates. Webix UI is created through templates as well. Updated sample ( https://github.com/webix-hub/webix-meteor-example ) uses this approach. The major disadvantage here that you need to have HTML placeholder for each Webix UI block that seriously limit Webix layout auto-sizing ability.
- "Webix first" app. I have published example at https://github.com/webix-hub/webix-meteor-mvc. This is a port of mvc layer from webix-adminapp-demo. Instead of using Meteor templates it uses Webix views as main building block. ( you still can use Meteor templates inside of webix views ) Such approach allows to fully utilize the functionality of Webix Layout, but it is quite different from common Meteor app.
